api/agent/agent.go

```diff
@@ -795,12 +703,15 @@ func (a *agent) runHotReq(ctx context.Context, call *call, state ContainerState,
 // and stderr can be swapped out by new calls in the container.  input and
 // output must be copied in and out.
 type container struct {
-	id      string // contrived
-	image   string
-	env     map[string]string
-	memory  uint64
-	cpus    uint64
-	timeout time.Duration // cold only (superfluous, but in case)
+	id     string // contrived
+	image  string
+	env    map[string]string
+	memory uint64
+	cpus   uint64
+
+	// limit total requests that can be run on this container
+	// default: zero - unlimited
+	maxRequests uint64
```

Comment:
conceptually, this doesn't seem like a 'container' construct wrt the docker api itself (this `*container`) as the `fsouza.DockerContainer` doesn't have any idea of this. can we push it up into `runHot` and build from the call state (it seems like a 'route' construct potentially, which we could hang off the call)